ai_edge_torch/generative/doc/system_overview.md

```diff
@@ -0,0 +1,128 @@
+# AI Edge Torch Generative API System Architecture Overview
+
+This doc aims to provide a technical deep dive of the AI Edge Torch Generative API, discuss its design considerations, system architecture and major components, current limitations, and future plans for improved system health, usability and performance.
+
+## Main objective
+
+As LLMs/LGMs gain its popularity, we are seeing an increased adoption of this technology in various forms of applications, from serving LLMs in the cloud to deploying smaller language models(SLM) on-device. Running generative AI models on-device has benefits such as better privacy protection, faster response time and low maintenance cost. There is often lots of optimization work to be done to make LLMs run efficiently on-device, which may involve complex methods such as model quantization, numerical debugging, model quality evaluation, performance optimizations and end-to-end integration with the production environment (e.g. Android/iOS apps, or web). To overcome those issues, we introduced the AI Edge Torch Generative API, which aims to provide a complete end-to-end solution to solve the deployment painpoints. The library is still under active development and the API interface may change in the coming months.
```

Comment:
nit: This can be re-written to read better. I generally ask Gemini to take care of that for me and verify the generated text.

Reply:
done